File: packages/go-sdk/operations/webhooks.go
package operations

import "context"

// WebhooksService provides webhook management operations.
type WebhooksService struct{ r Requester }

func NewWebhooksService(r Requester) *WebhooksService { return &WebhooksService{r: r} }

func (s *WebhooksService) ListSubscriptions(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/webhooks/subscriptions", query, nil, nil, &result)
	return result, err
}

func (s *WebhooksService) CreateSubscription(ctx context.Context, body any) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", "/v1/webhooks/subscriptions", nil, nil, body, &result)
	return result, err
}

func (s *WebhooksService) DeleteSubscription(ctx context.Context, id string) error {
	return s.r.DoRequest(ctx, "DELETE", PathParams("/v1/webhooks/subscriptions/{id}", map[string]string{"id": id}), nil, nil, nil, nil)
}

func (s *WebhooksService) ListDeliveries(ctx context.Context, query map[string]string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", "/v1/webhooks/deliveries", query, nil, nil, &result)
	return result, err
}

func (s *WebhooksService) GetDelivery(ctx context.Context, id string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "GET", PathParams("/v1/webhooks/deliveries/{id}", map[string]string{"id": id}), nil, nil, nil, &result)
	return result, err
}

func (s *WebhooksService) RetryDelivery(ctx context.Context, id string) (map[string]any, error) {
	var result map[string]any
	err := s.r.DoRequest(ctx, "POST", PathParams("/v1/webhooks/deliveries/{id}/retry", map[string]string{"id": id}), nil, nil, nil, &result)
	return result, err
}

File: packages/rust-sdk/src/operations/webhooks.rs
use crate::client::StraitClient;
use crate::errors::StraitError;
use crate::http::substitute_path_params;
use std::sync::Arc;

pub struct WebhooksService {
    client: Arc<StraitClient>,
}

impl WebhooksService {
    pub fn new(client: Arc<StraitClient>) -> Self {
        Self { client }
    }

    pub async fn list_subscriptions(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/webhooks/subscriptions", query, None, None).await
    }

    pub async fn create_subscription(&self, body: serde_json::Value) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("POST", "/v1/webhooks/subscriptions", None, None, Some(body)).await
    }

    pub async fn delete_subscription(&self, id: &str) -> Result<(), StraitError> {
        let path = substitute_path_params("/v1/webhooks/subscriptions/{id}", &[("id", id)]);
        self.client.do_request_no_content("DELETE", &path, None, None, None).await
    }

    pub async fn list_deliveries(&self, query: Option<&[(&str, &str)]>) -> Result<serde_json::Value, StraitError> {
        self.client.do_request("GET", "/v1/webhooks/deliveries", query, None, None).await
    }

    pub async fn get_delivery(&self, id: &str) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/webhooks/deliveries/{id}", &[("id", id)]);
        self.client.do_request("GET", &path, None, None, None).await
    }

    pub async fn retry_delivery(&self, id: &str) -> Result<serde_json::Value, StraitError> {
        let path = substitute_path_params("/v1/webhooks/deliveries/{id}/retry", &[("id", id)]);
        self.client.do_request("POST", &path, None, None, None).await
    }
}
